bundles/extension.rst

```diff
@@ -147,3 +147,44 @@ the full classmap executing the ``dump-autoload`` command of Composer.
     This technique can't be used when the classes to compile use the ``__DIR__``
     or ``__FILE__`` constants, because their values will change when loading
     these classes from the ``classes.php`` file.
+
+Loading Services directly in your Bundle class
+----------------------------------------------
+
+.. versionadded:: 6.1
+
+    The ``AbstractBundle`` class is introduced in Symfony 6.1.
+
+Alternatively, you can define and load services configuration directly in a bundle class
+by extending from the :class:`Symfony\\Component\\HttpKernel\\Bundle\\AbstractBundle`
+and defining the :method:`Symfony\\Component\\HttpKernel\\Bundle\\AbstractBundle::loadExtension` method::
+
+    use Symfony\Component\HttpKernel\Bundle\AbstractBundle;
+    use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;
+
+    class AcmeFooBundle extends AbstractBundle
+    {
+        public function loadExtension(array $config, ContainerConfigurator $container, ContainerBuilder $builder): void
```

Comment:
This comment is about the feature itself (maybe too late now), but how getting the builder here can be useful? What can be done with it that cannot be done with the configurator?

Reply:
The `ContainerBuilder` contains still useful methods that are not included in the `ContainerConfigurator`, for example, `removeDefinition()`, `hasDefinition()`, `registerForAutoconfiguration()`, `prependExtensionConfig()`, `getExtensionConfig()`, etc. (the last two useful in the new `prependExtension()`) 

In this method, we used to import the service definitions from an external file, and then we can remove some if desired using the builder instance. That's why we keep both.

Reply:
I see, thank you for your explanation :).